src/Cache.cpp

```diff
@@ -169,13 +169,36 @@ Cache::setup()
 
         nhlog::db()->debug("setting up cache");
 
+        // Previous location of the cache directory
+        auto oldCache = QString("%1/%2%3")
+                          .arg(QStandardPaths::writableLocation(QStandardPaths::CacheLocation))
+                          .arg(QString::fromUtf8(localUserId_.toUtf8().toHex()))
+                          .arg(QString::fromUtf8(settings->profile().toUtf8().toHex()));
+
         cacheDirectory_ = QString("%1/%2%3")
-                            .arg(QStandardPaths::writableLocation(QStandardPaths::CacheLocation))
+                            .arg(QStandardPaths::writableLocation(QStandardPaths::AppDataLocation))
                             .arg(QString::fromUtf8(localUserId_.toUtf8().toHex()))
                             .arg(QString::fromUtf8(settings->profile().toUtf8().toHex()));
 
         bool isInitial = !QFile::exists(cacheDirectory_);
 
+        // NOTE: If both cache directories exist it's better to do nothing: it
+        // could mean a previous migration failed or was interrupted.
+        bool needsMigration = isInitial && QFile::exists(oldCache);
+
+        if (needsMigration) {
+                nhlog::db()->info("found old state directory, migrating");
+                if (!QDir().rename(oldCache, cacheDirectory_)) {
```

Comment:
Does this work, if the parent directory does not exist?

Reply:
I think so, but that can't happen because the DataLocation is created during nheko initialisation, in main.cpp.

Reply:
Oh, I missed that, makes sense!